File: windows/Scanner/Detectors/CodexCliDetector.cs
using AIExposureScanner.Scanner.Filesystem;
using AIExposureScanner.Scanner.Models;
using AIExposureScanner.Scanner.Parsing;

namespace AIExposureScanner.Scanner.Detectors;

public sealed class CodexCliDetector : IDetector
{
    public string Id => "codex-cli";
    public string DisplayName => "Codex CLI";

    public ScanFacts CollectFacts(IFilesystemFacade fs)
    {
        var configPath = DetectorSupport.HomePath(fs, ".codex/config.toml");
        var authPath = DetectorSupport.HomePath(fs, ".codex/auth.json");
        var facts = new ScanFacts();
        DetectorSupport.AppendAppInstallationFact(
            facts,
            fs,
            Id,
            [DetectorSupport.HomePath(fs, ".codex")]
        );

        var text = DetectorSupport.ReadIfPresent(fs, configPath);
        if (text is not null)
        {
            facts.McpServers.AddRange(FactParsers.ParseCodexToml(text, Id, configPath));
            DetectorSupport.AppendConfigFileFact(facts, fs, Id, configPath);
        }

        if (fs.FileExists(authPath))
        {
            facts.AuthFiles.Add(new AuthFileFact(Id, authPath));
        }

        return facts;
    }
}

File: macos/Scanner/Sources/Scanner/Detectors/CodexCliDetector.swift
public struct CodexCliDetector: Detector {
    public let id = "codex-cli"
    public let displayName = "Codex CLI"

    public init() {}

    public func collectFacts(fs: any FilesystemFacade) throws -> ScanFacts {
        let configPath = homePath(fs, ".codex/config.toml")
        let authPath = homePath(fs, ".codex/auth.json")
        var facts = ScanFacts()
        appendAppInstallationFact(
            &facts,
            fs: fs,
            appId: id,
            candidatePaths: [homePath(fs, ".codex")]
        )

        if let text = try readIfPresent(fs, path: configPath) {
            facts.mcpServers.append(contentsOf: FactParsers.parseCodexToml(text, appId: id, configPath: configPath))
            appendConfigFileFact(&facts, fs: fs, appId: id, path: configPath)
        }

        if fs.fileExists(authPath) {
            facts.authFiles.append(AuthFileFact(appId: id, filePath: authPath))
        }

        return facts
    }
}
